mdb-realm/src/deployApps.ts

```diff
@@ -1,14 +1,30 @@
 import * as core from "@actions/core";
 import * as fs from "fs";
-import { configureRealmCli, publishApplication, createCluster, waitForClusterDeployment, getConfig } from "./helpers";
+import {
+    configureRealmCli,
+    publishApplication,
+    createCluster,
+    waitForClusterDeployment,
+    getConfig,
+    getClusters,
+} from "./helpers";
 import path from "path";
+import { EnvironmentConfig } from "./config";
 
 async function run(): Promise<void> {
     try {
         const config = getConfig();
-
         const appsPath = core.getInput("appsPath", { required: false });
 
+        await deployCluster(config, appsPath);
+    } catch (error: any) {
+        core.setFailed(`An unexpected error occurred: ${error.message}\n${error.stack}`);
+    }
+}
+
+export async function deployCluster(config: EnvironmentConfig, appsPath: string): Promise<void> {
+    const clusterNames = config.useExistingCluster ? await getClusters(config) : [];
+    if (!clusterNames.includes(config.clusterName)) {
```

Comment:
Not sure I understand why that was moved to a different function.

Reply:
In order to be able to be called by the tests also.